src/dmd/mtype.d

```diff
@@ -481,7 +380,60 @@ alias MOD = ubyte;
  */
 extern (C++) abstract class Type : RootObject
 {
-    TY ty;
+    enum Kind : ubyte
+    {
+        array,     // slice array, aka T[]
+        staticArray,      // static array, aka T[dimension]
+        associativeArray, // associative array, aka T[type]
+        pointer,
+        reference,
+        function_,
+        ident,
```

Comment:
This is shortened as well, but I don't mind it.

Reply:
Let's get'r done.  I think that should be `identifier`.  The fact that I'm not sure is justification.

Reply:
`identifier` is shortened to `ident` in other places in the code.

Reply:
Done.